crates/intermediate-theorem-format/src/data/structure.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Asymptotic Labs
+// SPDX-License-Identifier: Apache-2.0
+
+//! Struct IR data structures
+
+use crate::data::types::TheoremType;
+use crate::data::Dependable;
+use crate::TheoremModuleID;
+
+/// Unique identifier for a struct in the program
+pub type TheoremStructID = usize;
+
+// ============================================================================
+// Struct IR
+// ============================================================================
+
+/// A struct definition
+#[derive(Debug, Clone)]
+pub struct TheoremStruct {
```

Comment:
I would not use `Theorem` in any of the names here, it's already implicit. Just `Struct`

Reply:
I don't think so, we can use their qualified names to avoid confusion

Reply:
Okay, I'll remove the prefix from everything then.